sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BatchScanExec.scala

```diff
@@ -132,4 +132,10 @@ case class BatchScanExec(
     val result = s"$nodeName$truncatedOutputString ${scan.description()} $runtimeFiltersString"
     redact(result)
   }
+
+  /**
+   * Returns the name of this type of TreeNode.  Defaults to the class name.
+   * Note that we remove the "Exec" suffix for physical operators here.
+   */
+  override def nodeName: String = s"BatchScan ${scan.name()}"
```

Comment:
Can we simply use v2 `Table.name`?

Reply:
I don't think `Scan` can provide more information than `Table` for UI node name purposes. Even the time-travel version is also available in `Table`.

Reply:
>  Can we simply use v2 `Table.name`?

Hi @cloud-fan - thank you for your comment, can you please tell me how can we use v2 `Table#name()` here?
`Scan` or `BatchScanExec` does not have a reference to `Table`. Thus I decided to use `Scan#name()`. 

I would be happy to publish a patch that uses v2 `Table#name()` if you can please tell me how it can be accessed.

Reply:
Can we refactor the code a bit and pass `Table#name` when constructing `BatchScanExec`? I think it's better than adding a new public API.

Reply:
Sure - let me check

Reply:
any updates?

Reply:
Sorry - I was out sick.

I was able to pass the `Table` reference to `BatchScanExec` and use `Table#name` with minor code refactoring.
I will test it with Iceberg's implementation of DSv2 and post an update soon :)

Reply:
Thank you for the update, @sumeetgajjar . Take care!

Reply:
Hi @cloud-fan - opened a new PR #37505 with the above-propsed implementation.
Please review it at your convenience.